src/lib/forms/Dropzone.svelte

```diff
@@ -1,28 +1,29 @@
 <script lang="ts">
-	export let value: string = '';
-	export let divClass: string = 'flex justify-center items-center w-full';
-	export let labelClass: string =
-		'flex flex-col justify-center items-center w-full h-64 bg-gray-50 rounded-lg border-2 border-gray-300 border-dashed cursor-pointer dark:hover:bg-bray-800 dark:bg-gray-700 hover:bg-gray-100 dark:border-gray-600 dark:hover:border-gray-500 dark:hover:bg-gray-600';
-	export let innerDivClass: string = 'flex flex-col justify-center items-center pt-5 pb-6';
+    export let id: string = 'dropzone-file';
+    export let value: string = '';
+    export let containerClass: string = 'flex justify-center items-center w-full';
+    export let labelClass: string =
+        'flex flex-col justify-center items-center w-full h-64 bg-gray-50 rounded-lg border-2 border-gray-300 border-dashed cursor-pointer dark:hover:bg-bray-800 dark:bg-gray-700 hover:bg-gray-100 dark:border-gray-600 dark:hover:border-gray-500 dark:hover:bg-gray-600';
+    export let innerDivClass: string = 'flex flex-col justify-center items-center pt-5 pb-6';
+    let input: HTMLInputElement;
 </script>
 
-<div class={divClass}>
-	<label for="dropzone-file" class={labelClass}>
-		<div class={innerDivClass}>
-			<slot />
-		</div>
-		<input
-			{...$$restProps}
-			bind:value
-			type="file"
-			class="hidden"
-			on:click
-			on:change
-			on:focus
-			on:blur
-			on:mouseenter
-			on:mouseleave
-			on:mouseover
-		/>
-	</label>
-</div>
+<button class={containerClass} on:click={() => input.click()}>
+    <label
+        class={labelClass}
+        for={id}
+        on:focus
+        on:blur
+        on:mouseenter
+        on:mouseleave
+        on:mouseover
+        on:dragenter
+        on:dragleave
+        on:dragover
+        on:drop>
+        <div class={innerDivClass}>
+            <slot />
+        </div>
+        <input {...$$restProps} bind:value bind:this={input} {id} type="file" class="hidden" on:change />
```

Comment:
I fixed some more issues I found while using the Dropzone. The hidden input is not able to recognize the drag events. So instead we listen on the label. Furthermore, a label/input/ div are not focusable as they are non-interactive elements / hidden. Wrapping the component in a button enables the user to `Tab` to the item. 
Controversial is that we would need to forward the click to the input. Let me know what you think.